[Flaky] k8s.io/kubernetes/pkg/api/testing

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

#### Which jobs are flaking:

pull-kubernetes-unit

#### Which test(s) are flaking:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/99817/pull-kubernetes-unit/1367685781799309312

```
=== FAIL: pkg/api/testing  (0.00s)
FAIL	k8s.io/kubernetes/pkg/api/testing	240.331s
```

Two new tests were just added in https://github.com/kubernetes/kubernetes/pull/99012 that take significant amounts of time in CI:

```
--- FAIL: TestUnstructuredRoundTripApplyConfigurations (112.70s)
--- FAIL: TestJsonRoundTripApplyConfigurations (35.82s)
```

~These tests also appear to be flaking regularly: https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&job=kubernetes-unit&test=k8s.io%2Fkubernetes%2Fpkg%2Fapi%2Ftesting~

/assign @jpbetz

--- a/pkg/api/testing/applyconfiguration_test.go
+++ b/pkg/api/testing/applyconfiguration_test.go
@@ -52,7 +52,7 @@ func TestUnstructuredRoundTripApplyConfigurations(t *testing.T) {
 		}
 
 		t.Run(gvk.String(), func(t *testing.T) {
-			for i := 0; i < 50; i++ {
+			for i := 0; i < 3; i++ {
 				item := fuzzObject(t, gvk)
 				builder := applyconfigurations.ForKind(gvk)
 				unstructuredRoundTripApplyConfiguration(t, item, builder)
@@ -80,7 +80,7 @@ func TestJsonRoundTripApplyConfigurations(t *testing.T) {
 		}
 
 		t.Run(gvk.String(), func(t *testing.T) {
-			for i := 0; i < 50; i++ {
+			for i := 0; i < 3; i++ {
 				item := fuzzObject(t, gvk)
 				builder := applyconfigurations.ForKind(gvk)
 				jsonRoundTripApplyConfiguration(t, item, builder)

